File: solutions/0746. Min Cost Climbing Stairs/0746.cs
// Time complexity: O(n)
// Space complexity: O(1)
public class Solution {
    public int MinCostClimbingStairs(int[] cost) {
        int prev = cost[0];
        int cur = cost[1];

        for (int i = 2; i < cost.Length; ++i) {
            int next = cost[i] + Math.Min(prev, cur);
            prev = cur;
            cur = next;
        }
        return Math.Min(prev, cur);
    }
}

File: solutions/0746. Min Cost Climbing Stairs/0746.js
// Time complexity: O(n)
// Space complexity: O(1)
var minCostClimbingStairs = function(cost) {
    let prev = cost[0];
    let cur = cost[1];

    for (let i = 2; i < cost.length; ++i) {
        let next = cost[i] + Math.min(prev, cur);
        prev = cur;
        cur = next;
    }
    return Math.min(prev, cur);
};